### ISSUE TITLE
Make `handle` method optional

### ISSUE THREAD

**Issue by imliam** [lorisleiva/laravel-actions#39]
Currently, the `handle()` method is required on an action - if it is not present, it will cause a `BadMethodCallException` when trying to use it.

My use case for making this optional is quite simple; it allows actions to be used as a complete replacement for controllers, even if you don't have any logic to handle in the action itself. An action could instead just define a `response()` method that returns a view or resource.

While the `handle()` method is required, there are generally 2 options to go, both of which make sense but aren't desirable when keeping things:

* Use a controller (or `Route::view()` if there is no extra data/validation/authorisation/etc. going into rendering it) to render it, which puts it in a different namespace and structure as the rest of your code
* Define an empty `handle()` method on the action, which works but is redundant

**Comment by mortenscheel** [comment]
Personally I think one of the biggest advantages of this package is the unified `handle()` method which can handle (pun intended) some logic no matter what context it's used in.

What exactly do you want to achieve? If you want your logic in a method with a different name, you can just create a ControllerAction class which extends Action, and use that in stead.

**Comment by imliam** [comment]
My point is that not everything necessarily has logic to be handled. If you want to display a view (ie. using an action as a controller) but gate it behind some authorisation/validation without any actual logic to be handled, actions are perfect for that.

The "fix" would purely be to have an empty handle method on the base action class that could be overridden if you did have logic to handle (which yes, a custom `ControllerAction` class in an app could do too). It wouldn't make the handle method obsolete or make it less unified.

**Comment by lorisleiva** [comment]
Hi there 👋 

Thanks for bringing this to my attention.

Usually, when I want an Action to simply returns a view I define it directly in the `handle` method since I know for sure it's not going to be used as anything different than a controller.

```php
class MyAction extends Action
{
    public function handle()
    {
        return view('my-action');
    }
}
```

However, it might be more elegant to keep the consistency of defining routes in the `response` method throughout the application.

The only issue I can potentially foresee is when running the action as a listener it will require a `handle` method. And, because of dependency injections, I cannot provide a `handle` method on the `Action` abstract class otherwise PHP will complain about signature mismatches.

I'll work on it this weekend and see if I can sort something out. Worst-case scenario it will be optional for everything but listeners. 😅

**Comment by lorisleiva** [comment]
🚀 Implemented and released in [`v1.0.2`](https://github.com/lorisleiva/laravel-actions/releases/tag/v1.0.2).
